packages/terminal/src/browser/terminal-widget-impl.ts

```diff
@@ -47,8 +47,11 @@ export interface TerminalWidgetFactoryOptions extends Partial<TerminalWidgetOpti
 export class TerminalWidgetImpl extends TerminalWidget implements StatefulWidget {
 
     private readonly TERMINAL = 'Terminal';
+    isInteractive = true;
+    terminalId = -1;
     protected readonly onTermDidClose = new Emitter<TerminalWidget>();
-    protected terminalId = -1;
+    protected readonly onTermDidDispose = new Emitter<void>();
```

Comment:
When do you dispose new emitters? Please use `this.toDispose.push`.

Reply:
i disposed them in the `dispose()` method. and you are right, it is cleaner and more organized to use toDispose.push()